readthedocs/doc_builder/environments.py

```diff
@@ -295,47 +285,37 @@ def __init__(self, project, environment=None):
         self.environment = environment or {}
         self.commands = []
 
-    def record_command(self, command):
+    def pre_run_command(self, cls, cmd, warn_only, kwargs):
+        """
+        Method to be called before the command is executed.
+
+        The command that will be executed can be accessed as
+        the last element of ``self.commands``.
+        """
         pass
 
-    def _log_warning(self, msg):
-        log.warning(LOG_TEMPLATE.format(
-            project=self.project.slug,
-            version='latest',
-            msg=msg,
-        ))
+    def post_run_command(self):
+        """
+        Method to be called after the command is executed.
+
+        The command that was executed can be accessed as
+        the last element of ``self.commands``.
+        """
+        pass
 
     def run(self, *cmd, **kwargs):
         """Shortcut to run command from environment."""
         return self.run_command_class(cls=self.command_class, cmd=cmd, **kwargs)
 
-    def run_command_class(
-            self, cls, cmd, record=None, warn_only=False,
-            record_as_success=False, **kwargs):
+    def run_command_class(self, cls, cmd, warn_only=False, **kwargs):
         """
         Run command from this environment.
 
         :param cls: command class to instantiate a command
         :param cmd: command (as a list) to execute in this environment
-        :param record: whether or not to record this particular command
-            (``False`` implies ``warn_only=True``)
         :param warn_only: don't raise an exception on command failure
-        :param record_as_success: force command ``exit_code`` to be saved as
```

Comment:
We are lossing the documentation for these attributes here. It should be added where it fit.